examples/build_file_generation/WORKSPACE

```diff
@@ -28,45 +36,97 @@ http_archive(
     ],
 )
 
+# Load rules_go ruleset and expose the toolchain and dep rules
 load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies")
 load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
 
+# go_rules_dependencies is a function that registers external dependencies
+# needed by the Go rules.
+# https://github.com/bazelbuild/rules_go/blob/master/go/dependencies.rst#go_rules_dependencies
 go_rules_dependencies()
 
+# go_rules_dependencies is a function that registers external dependencies
+# needed by the Go rules.
+# https://github.com/bazelbuild/rules_go/blob/master/go/dependencies.rst#go_rules_dependencies
 go_register_toolchains(version = "1.18.3")
 
+# The following call configured the gazelle dependencies, Go environment and Go SDK.
 gazelle_dependencies()
 
-######################################################################
 # Remaining setup is for rules_python
 
+# You do not want to use the following command when you are using a WORKSPACE file
+# that is outside of rules_python repository.
+# This command allows targets from a local directory to be bound.
+# Which allows bazel to use targets defined in base rules_python directory.
+# If you are using this example outside of the rules_python git repo,
+# use the http_archive command that is commented out below.
+# https://bazel.build/reference/be/workspace#local_repository
 local_repository(
     name = "rules_python",
     path = "../..",
 )
 
+# When not using this example in the rules_python git repo you would load the python
+# ruleset using the following StarLark.
+# See https://github.com/bazelbuild/rules_python#getting-started for the latest
+# ruleset version.
+#
+# The following StarLark would replace the `local_repository` rule mentioned above.
+#
+# rules_python_version = "740825b7f74930c62f44af95c9a4c1bd428d2c53" # Latest @ 2021-06-23
```

Comment:
I think it's better to pin to a release version instead of pointing to a specific commit on the main branch. Automation tools that bump dependencies often will try to always bump to the latest commit on a branch, which is not correct for production usage (in general).

Reply:
The examples are used as integration tests, so we have to keep the local_repository there. We could replace it on CI, but we don't have this feature now.

Maybe on the local_repository, add a comment pointing to the latest release on GitHub, which is a static address.